lib/hanami/application/settings.rb

```diff
@@ -1,22 +1,94 @@
 # frozen_string_literal: true
 
+require "dry/configurable"
 require "dry/core/constants"
-require_relative "settings/definition"
-require_relative "settings/struct"
 
 module Hanami
   class Application
     # Application settings
     #
+    # Users are expected to inherit from this class to define their application
+    # settings.
+    #
+    # @example
+    #   # config/settings.rb
+    #   # frozen_string_literal: true
+    #
+    #   require "hanami/application/settings"
+    #   require "my_app/types"
+    #
+    #   module MyApp
+    #     class Settings < Hanami::Application::Settings
+    #       setting :database_url
+    #       setting :feature_flag, default: false, constructor: Types::Params::Bool
+    #     end
+    #   end
+    #
+    # Settings are defined with
+    # [dry-configurable](https://dry-rb.org/gems/dry-configurable/), so you can
+    # take a look there to see the supported syntax.
+    # 
+    # Users work with an instance of this class made available within the
+    # `settings` key in the container. The instance gets its settings populated
+    # from a configurable store, which defaults to
+    # {Hanami::Application::Settings::DotenvStore}. It uses
+    # [dotenv](https://github.com/bkeepers/dotenv) under the hood. The usual
```

Comment:
I'd not explain which `.env` files are supported, but just link via YARD tag `@see Hanami::Application::Settings::DotenvStore`. As a reader, there I expect a fully explanation of the `.env` files, including their priority.

Reply:
Yeah, I put everything here because `DotenvStore` is a kind of private API, but probably it makes more sense to do it in that way. I'll update it and let you know.

Reply:
Done @jodosha!